main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -242,9 +238,13 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       SP ~ Names.QualifiedClass ~ optWS ~ "[" ~ optWS ~ Type ~ optWS ~ "]" ~ SP ~> ParsedAst.TypeConstraint
     }
 
+    def OptTypeConstraintList: Rule1[Seq[ParsedAst.TypeConstraint]] = rule {
+      optional(optWS ~ keyword("with") ~ optWS ~ oneOrMore(TypeConstraint).separatedBy(optWS ~ "," ~ optWS)) ~> ((o: Option[Seq[ParsedAst.TypeConstraint]]) => o.getOrElse(Seq.empty))
```

Comment:
Should it really be optWS here? Would that not allow you to write e.g. `withEq[a]`??

Reply:
`keyword` has a lookahead that prevents this. But could change to `WS` for consistency.

Reply:
But what about the `optWS` before? I think we should put hard WS in both places.

Reply:
Can you explore what happened? If you cannot find anything then we can just merge it.

Reply:
I don't know if we should be so strict when there's no ambiguity.

Scala allows e.g. `class C[a]extends T` (no space before `extends`)

Reply:
Ah- but we don't allow such things anywhere currently.

Reply:
Fair enough. Fixed.